app/src/main/java/com/sci/recipeandroid/feature/personalize/ui/viewmodel/PersonalizeViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package com.sci.recipeandroid.feature.personalize.ui.viewmodel
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.sci.recipeandroid.feature.personalize.domain.repository.PersonalizeRepository
+import com.sci.recipeandroid.feature.personalize.domain.model.PersonalizeDataModel
+import kotlinx.coroutines.launch
+
+class PersonalizeViewModel(private val personalizeRepository: PersonalizeRepository) : ViewModel() {
+
+    private val _liveData = MutableLiveData<PersonalizeUiState>()
+
+    val liveData: LiveData<PersonalizeUiState> = _liveData
+
+
+
+    init {
+        _liveData.value = PersonalizeUiState.Loading
+        viewModelScope.launch {
```

Comment:
viewModel only should know Result return from data layer , plz try to use with Result class instead of try catch.

Reply:
Updated in a3e2aa4814dc66a3b2a70289d7b8af5b4ae2bc9b